packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx

```diff
@@ -9,9 +9,11 @@ import {
 } from './searchFacetInterfaces';
 
 export class SearchFilterTransform {
-  static SPLIT_UNQUOTED_COMMAS = /,(?=(?:(?:[^"\\"]*["\\"]){2})*[^"\\"]*$)/;
-  static SPLIT_UNQUOTED_COLONS = /:(?=(?:(?:[^"\\"]*["\\"]){2})*[^"\\"]*$)/;
-  static SPLIT_UNQUOTED_PIPES = /\|(?=(?:(?:[^"\\"]*["\\"]){2})*[^"\\"]*$)/;
+  static SPLIT_UNESCAPED_COMMAS = /(?<!\\),/;
+  static SPLIT_UNESCAPED_COLONS = /(?<!\\):/;
+  static SPLIT_UNESCAPED_PIPES = /(?<!\\)\|/;
+  static CHARS_TO_ESCAPE = /[|",:()[\]<>^*~.!]/g;
+  static ESCAPED_CHAR = /\\([|",:()[\]<>^*~.!])/g;
```

Comment:
Ugh, regexes are awful and unmaintainable!

I suggest adding comments to these, specifically with the intent of the regexes (seems like there's two groups: the SPLIT_ and the CHAR ones -- comment for each group at least).

Reply:
Agreed, I had actually committed, but not yet pushed comments for these before you reviewed. 😄